easybuild/easyblocks/h/healpix.py

```diff
@@ -112,20 +135,42 @@ def install_step(self):
 
+    def make_module_extra(self):
+        """additional paths"""
+        txt = super(EB_HEALPix, self).make_module_extra()
+        paths = {
+            'CPATH': 'include',
+            'LIBRARY_PATH': 'lib',
+            'PATH': 'bin',
+        }
+        for key in sorted(paths):
+            txt += self.module_generator.prepend_paths(key, os.path.join('src', 'cxx', self.target_string, paths[key]))
+
+        return txt
+
     def sanity_check_step(self):
-        """Custom sanity check for HEALPix < 3.50"""
-        # since 3.50 files changed, so we specify it in easyconfig.
-        # "else" section below is for backward-compatibility, ATM only easyconfig for 2.20a-icte is known
-        if LooseVersion(self.version) >= LooseVersion('3.50'):
-            super(EB_HEALPix, self).sanity_check_step()
-        else:
-            custom_paths = {
-                'files': [os.path.join('bin', x) for x in ['alteralm', 'anafast', 'hotspot', 'map2gif', 'median_filter',
-                                                           'plmgen', 'sky_ng_sim', 'sky_ng_sim_bin', 'smoothing',
-                                                           'synfast', 'ud_grade']] +
-                         [os.path.join('lib', 'lib%s.a' % x) for x in ['chealpix', 'gif', 'healpix', 'hpxgif',
-                                                                       'psht_healpix_f']] +
-                         [os.path.join('lib', 'libchealpix.%s' % get_shared_lib_ext())],
-                'dirs': [os.path.join('include')],
-            }
-            super(EB_HEALPix, self).sanity_check_step(custom_paths=custom_paths)
+        """Custom sanity check for HEALPix."""
+        binaries = [os.path.join('bin', x) for x in ['alteralm', 'anafast', 'hotspot', 'map2gif', 'median_filter',
+                                                     'plmgen', 'sky_ng_sim', 'sky_ng_sim_bin', 'smoothing',
+                                                     'synfast', 'ud_grade']]
+        libraries = [os.path.join('lib', 'lib%s.a' % x) for x in ['chealpix', 'gif', 'healpix', 'hpxgif']]
+
+        target_subdirs = ['bin', 'lib']
+        if LooseVersion(self.version) >= LooseVersion('3'):
+            # 'include' subdir is only there for recent HEALPix versions
+            target_subdirs.append('include')
+
+        custom_paths = {
+            'files': binaries + libraries + [os.path.join('lib', 'libchealpix.%s' % get_shared_lib_ext())],
+            'dirs': ['include'] + [os.path.join('src', 'cxx', self.target_string, x) for x in target_subdirs],
+        }
+
+        custom_commands = []
+        if LooseVersion(self.version) >= LooseVersion('3'):
+            custom_commands.extend([
+                "cd %s && make test &> build/testresults.txt" % self.installdir,
+                'test $(grep -c "test completed" %s/build/testresults.txt) -eq 4' % self.installdir,
+                'test $(grep -c "normal completion" %s/build/testresults.txt) -eq 10' % self.installdir,
```

Comment:
@jose-d How about loosening this up a little bit by using `-gt` rather than `-eq`?

If more tests pass or complete normally, there's no problem?

Reply:
my opinion: when there is change in test suite "test completed" and "normal completion" counts, we can expect major change of testsuite. So far I have seen non-conservative changes in whole Healpix project across versions, testsuite is returning '0' RC even when failing, and minor issues are not changing the final summary. 

I'd propose to keep this as it is, and possibly in next version:
a) patch the testsuite to behave like in mainstream projects = nonzero return code when fail
b) move magic constants "4" and "10" into easyconfig if we'll find them heavy-version specific
c) something better

Reply:
Thanks for clarifying @jose-d!